### PR TITLE
Add JS RegExp compat data

### PR DESCRIPTION
**Pull request by Elchi3**

### CODE REVIEW

File: javascript/builtins/RegExp.json [hunk 1]
```diff
@@ -0,0 +1,1956 @@
...
+              },
+              "opera_android": {
+                "version_added": true
+              },
+              "safari": {
+                "version_added": true
+              },
+              "safari_ios": {
+                "version_added": true
+              }
```
Comment (teoli2003) [inline javascript/builtins/RegExp.json]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/flags indicates no support for Opera and Android. Was the change of false -> true done for these done on purpose?

File: javascript/builtins/RegExp.json [hunk 2]
```diff
@@ -0,0 +1,1956 @@
...
+              "description": "\"(?:)\" for empty regexps",
+              "support": {
+                "webview_android": {
+                  "version_added": null
+                },
+                "chrome": {
+                  "version_added": null
+                },
+                "chrome_android": {
+                  "version_added": null
```
Comment (teoli2003) [inline javascript/builtins/RegExp.json]: This was explicitely false in the original table. Has the change to null been done on purpose here?

File: javascript/builtins/RegExp.json [hunk 3]
```diff
@@ -0,0 +1,1956 @@
...
+                },
+                "opera": {
+                  "version_added": null
+                },
+                "opera_android": {
+                  "version_added": null
+                },
+                "safari": {
+                  "version_added": null
```
Comment (teoli2003) [inline javascript/builtins/RegExp.json]: This was true in the original table, has the change to null been done on purpose?

File: javascript/builtins/RegExp.json [hunk 4]
```diff
@@ -0,0 +1,1956 @@
...
+              }
+            }
+          }
+        },
+        "sticky": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/RegExp/sticky",
+            "support": {
+              "webview_android": {
+                "version_added": true
```
Comment (teoli2003) [inline javascript/builtins/RegExp.json]: It was false in the original table. Why true and not "49" if it is supported?

File: javascript/builtins/RegExp.json [hunk 5]
```diff
@@ -0,0 +1,1956 @@
...
+              }
+            },
+            "status": {
+              "experimental": false,
+              "standard_track": true,
+              "deprecated": false
+            }
+          },
+          "escaping": {
```
Comment (teoli2003) [inline javascript/builtins/RegExp.json]: Chrome, IE, Opera, and Safari where initially set to 'true', now to 'null'. Done on purpose?

File: javascript/builtins/RegExp.json [hunk 6]
```diff
@@ -0,0 +1,1956 @@
...
+                "version_added": true
+              },
+              "chrome": {
+                "version_added": "50"
+              },
+              "chrome_android": {
+                "version_added": "50"
+              },
+              "edge": {
+                "version_added": "12"
```
Comment (teoli2003) [inline javascript/builtins/RegExp.json]: A note for case folding in Edge 13?

File: javascript/builtins/RegExp.json (hunk 1 above)
Comment (Elchi3) [reply javascript/builtins/RegExp.json]: Yes, I think it is supported in these browsers by now.

File: javascript/builtins/RegExp.json (hunk 2 above)
Comment (Elchi3) [reply javascript/builtins/RegExp.json]: I just tested `new RegExp().source;` in Chrome and it returns the correct "(?:)". So, I'm changing this to true.

File: javascript/builtins/RegExp.json (hunk 4 above)
Comment (Elchi3) [reply javascript/builtins/RegExp.json]: Yes, 49 is better. Fixed

File: javascript/builtins/RegExp.json (hunk 5 above)
Comment (Elchi3) [reply javascript/builtins/RegExp.json]: Fixed

File: javascript/builtins/RegExp.json (hunk 6 above)
Comment (Elchi3) [reply javascript/builtins/RegExp.json]: Added.

**Comment by teoli2003** [comment]
So the only open question is if we should add RegExp.length, or not. If no, r+ (LGTM)

**Comment by Elchi3** [comment]
> So the only open question is if we should add RegExp.length, or not.

I think we shouldn't. It has no MDN page, and it is just a property with the value of "2".

Thanks for your review!

**Pull request merged**